### ISSUE TITLE
js error when installing service

### ISSUE THREAD

**Issue by dsbert** [jon-hall/pm2-windows-service#28]
I receive the following error when running `pm2-service-install`.

```
C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2-windows-service\node
_modules\promisify-node\utils\args.js:9
  var args = func.toString().match(/function\s.*?\(([^)]*)\)/)[1];
                                                              ^

TypeError: Cannot read property '1' of null
    at module.exports (C:\Users\Administrator\AppData\Roaming\npm\node_modules\p
m2-windows-service\node_modules\promisify-node\utils\args.js:9:63)
    at C:\Users\Administrator\AppData\Roaming\npm\node_modules\pm2-windows-servi
ce\node_modules\promisify-node\index.js:79:29
    at Array.filter (<anonymous>)
    at processExports (C:\Users\Administrator\AppData\Roaming\npm\node_modules\p
m2-windows-service\node_modules\promisify-node\index.js:60:6)
    at module.exports (C:\Users\Administrator\AppData\Roaming\npm\node_modules\p
m2-windows-service\node_modules\promisify-node\index.js:112:10)
    at Object.<anonymous> (C:\Users\Administrator\AppData\Roaming\npm\node_modul
es\pm2-windows-service\src\install.js:7:11)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
```

This is on node v 8.4.0

**Comment by musukvl** [comment]
See the same output for windows server 2016
```
C:\Program Files\Far Manager>npm i pm2-windows-service -g
C:\Users\dev\AppData\Roaming\npm\pm2-service-install -> C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\bin\pm2-service-install
C:\Users\dev\AppData\Roaming\npm\pm2-service-uninstall -> C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\bin\pm2-service-uninstall

> spawn-sync@1.0.15 postinstall C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\node_modules\spawn-sync
> node postinstall

+ pm2-windows-service@0.2.0
added 123 packages in 10.544s

C:\Program Files\Far Manager>pm2-service-install pm2service
C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\node_modules\promisify-node\utils\args.js:9
  var args = func.toString().match(/function\s.*?\(([^)]*)\)/)[1];
                                                              ^

TypeError: Cannot read property '1' of null
    at module.exports (C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\node_modules\promisify-node\utils\args.js:9:63)
    at C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\node_modules\promisify-node\index.js:79:29
    at Array.filter (<anonymous>)
    at processExports (C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\node_modules\promisify-node\index.js:60:6)
    at module.exports (C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\node_modules\promisify-node\index.js:112:10)
    at Object.<anonymous> (C:\Users\dev\AppData\Roaming\npm\node_modules\pm2-windows-service\src\install.js:7:11)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
```

**Comment by musukvl** [comment]
Issue happens only since node 8.3.0 for node 8.2.1 it works fine.

**Comment by firrae** [comment]
I have this same error and this is for a production server so this is a complete show stopper and makes me need to get our clients to swap from PM2 to something else which is going to be a major pain as they are installing now...

Windows Server 2016
Node 8.4.0
NPM 5.4.1

**Comment by ilusharulkov** [comment]
I have the same issue. On node v8.5 too

**Comment by jon-hall** [comment]
Hi, my apologies to everyone for not getting on to this sooner.

Thanks all for the detailed reports, especially for including the node versions - it seems to be only be a node 8 problem so far, so I fixed it using a node 8 only feature.

I think this should sort it for now and have published what should be a fixed version (`0.2.1`), if anyone does get anymore issues though (which might well still happen on node 7 and below), please reopen this ticket and I'll investigate further.
